src/Interfaces.ts

```diff
@@ -0,0 +1,60 @@
+import {
+    GraphQLSchema,
+    GraphQLFieldDefinition,
+    GraphQLResolveInfo,
+    GraphQLResult,
+    GraphQLType,
+} from 'graphql';
+
+/* TODO: Add documentation */
+
+export interface IResolverValidationOptions {
+    requireResolversForArgs?: boolean;
+    requireResolversForNonScalar?: boolean;
+    requireResolversForAllFields?: boolean;
+}
+
+export type ITypedef = (() => ITypedef[]) | string;
+export type ITypeDefinitions = ITypedef | ITypedef[];
+export type IResolveTypeFn = (data: any, context: any, info: GraphQLResolveInfo) => any;
+export type IResolveFn = (rootObject?: any, args?: { [key: string]: any }, context?: any, info?: GraphQLResolveInfo) => any;
+export type IResolverObject = { [key: string]: IResolveFn | {resolve?: IResolveFn, __resolveType?: IResolveTypeFn}};
+export interface IResolvers {
+    [key: string]: (() => any) | IResolverObject;
+};
+export interface ILogger {
+    log: (message: string | Error) => void;
+}
+
+export interface IConnectorCls {
```

Comment:
Let's get rid of connectors altogether. While the concept is useful, it doesn't make sense to have a special function in graphql-tools to attach them to the context. It's better to teach people how to do it manually.